testing/deployer/sprawl/catalog.go

```diff
@@ -318,13 +318,45 @@ func serviceToCatalogRegistration(
 			Address: node.LocalAddress(),
 		},
 	}
+	if svc.IsMeshGateway {
+		reg.Service.Kind = api.ServiceKindMeshGateway
+		reg.Service.Proxy = &api.AgentServiceConnectProxyConfig{
+			Config: map[string]interface{}{
+				"envoy_gateway_no_default_bind":       true,
+				"envoy_gateway_bind_tagged_addresses": true,
+			},
+			MeshGateway: api.MeshGatewayConfig{
+				Mode: api.MeshGatewayModeLocal,
+			},
+		}
+	}
 	if node.HasPublicAddress() {
 		reg.TaggedAddresses = map[string]string{
 			"lan":      node.LocalAddress(),
 			"lan_ipv4": node.LocalAddress(),
 			"wan":      node.PublicAddress(),
 			"wan_ipv4": node.PublicAddress(),
 		}
+		// TODO: not sure what the difference is between these, but with just the
```

Comment:
Was it being set in agentful mode?

Looking at the proxy config code I would expect node tagged addresses to generally be ignore for all proxies.

I _think_ the only way to use tagged addresses for service mesh is to set the services tagged addresses like you have done here.

Reply:
Compared with an agentful mgw in DC1, I see that it has both `TaggedAddresses` and `ServiceTaggedAddresses` filled. Compared with an agentful service node in DC1, only `TaggedAddresses`. Agentless service in DC2 has neither `(Service)TaggedAddresses`, only `Address` and `ServiceAddress` (which makes some sense since it only has a LAN connection).